spec/views/_dataset_spec.rb

```diff
@@ -33,6 +33,27 @@
     expect(page.css('tr')[0].css('td')[2].inner_text).to match(/Yes/)
   end
 
+  def expect_columns(page)
+    expect(page.css('tr')[0].css('td').count).to eq(8)
+  end
+
+  it 'displays warning icon for URL inaccessible dataset' do
+    deprecated_date = DateTime.now
+    @dataset.update_column(:url, "http://www.deadurl.com/example.csv")
+    @dataset.update_column(:url_deprecated_at, deprecated_date)
+    render :partial => 'datasets/dataset.html.erb', :locals => {:dataset => @dataset}
+    page = Nokogiri::HTML(rendered)
+    expect(page.css('tr:first-child > td:nth-child(2)')).to have_css('i.fa.fa-exclamation-triangle');
+  end
+
+  it 'displays deprecation date when in the dashboard' do
+    @dashboard = true
+    render :partial => 'datasets/dataset.html.erb', :locals => {:dataset => @dataset}
+    page = Nokogiri::HTML(rendered)
+    expect_columns(page)
+    expect(DateTime.parse(page.css('tr')[0].css('td')[5].inner_text).instance_of?(DateTime))
```

Comment:
So I've going to include timecop in that spec because I can see the utility of maybe being able to ensure that no deprecated dates in the dataset are older than 24 hours (for checking that the heroku cron job is functioning as anticipated. For now I'm not going to amend the above because the rails view returns the inner text of that field via `to_formatted_s(:short)`, so I think confirming that it's a datetime should suffice (although I hope that Ruby parsing bug gets fixed) and also involves the least additional code to do equality tests between what the view contains and what is stored in the database - sound sane?